verification/cbmc/sources/make_common_data_structures.c

```diff
@@ -79,13 +80,9 @@ void ensure_record_has_allocated_members(struct aws_cryptosdk_keyring_trace_reco
 
 void ensure_trace_has_allocated_records(struct aws_array_list *trace, size_t max_len) {
     /* iterate over each record in the keyring trace */
-    size_t num_records = aws_array_list_length(trace);
-    for (size_t idx = 0; idx < num_records; ++idx) {
-        struct aws_cryptosdk_keyring_trace_record *record;
-        if (!aws_array_list_get_at_ptr(trace, (void **)&record, idx)) {
-            /* make sure each record is valid */
-            ensure_record_has_allocated_members(record, max_len);
-        }
+    for (size_t i = 0; i < trace->length; ++i) {
+        struct aws_cryptosdk_keyring_trace_record *data = (struct aws_cryptosdk_keyring_trace_record *)trace->data;
```

Comment:
Could you move the declaration outside of the loop?

Reply:
Done!